src/shared/bpf-link.c

```diff
@@ -19,6 +20,18 @@ bool can_link_bpf_program(BpfProgram *prog) {
         return libbpf_get_error(link) == -EBADF;
 }
 
+int serialize_bpf_link(FILE *f, FDSet *fds, const char *key, BpfLink *link) {
+        int fd;
+
+        assert(key);
+        assert(link);
+        if (libbpf_get_error(link) != 0)
+                return -EINVAL;
```

Comment:
I merged this commit with the following one.
The reason why I separated `src/shared/bpf-link.[ch]` is to make that reusable in other PRs, but shouldn't matter much if PR gets merged ;)

Reply:
Wrt `libbpf_get_error` return value: it should be vetted,  e.g. some return codes may be libbpf specific, e.g. https://github.com/libbpf/libbpf/blob/master/src/libbpf.c#L7285
https://github.com/libbpf/libbpf/blob/master/src/libbpf.c#L9024

Also libbpf errors are logged to kernel log buffer via `pr_warn` so IMO it is not worthy to duplicate them in user space

Reply:
>  some return codes may be libbpf specific

That is a valid point.

> ibbpf errors are logged to kernel log buffer via pr_warn

This doesn't really matter. If there was a valid errno-style error code, we would want to propagate it. Since there isn't, this is moot. But the general rule is that error codes should be propagated as long as they make sense in the new context.